controllers/controllers/workloads/cfprocess_controller.go

```diff
@@ -240,11 +248,37 @@ func (r *CFProcessReconciler) fetchLRPsForProcess(ctx context.Context, cfProcess
 	return lrpsForProcess, err
 }
 
-func secretDataToEnvMap(secretData map[string][]byte) map[string]string {
+func (r *CFProcessReconciler) getPort(ctx context.Context, cfProcess workloadsv1alpha1.CFProcess, cfApp workloadsv1alpha1.CFApp) (int, error) {
+	// Get Routes for the process
+	var cfRoutesForProcess networkingv1alpha1.CFRouteList
+	err := r.Client.List(ctx, &cfRoutesForProcess, client.InNamespace(cfApp.GetNamespace()), client.MatchingFields{shared.DestinationAppName: cfApp.Name})
+	if err != nil {
+		return 0, err
+	}
+	// Filter those destinations
+	for _, cfRoute := range cfRoutesForProcess.Items {
+		for _, destination := range cfRoute.Status.Destinations {
+			if destination.ProcessType == cfProcess.Spec.ProcessType && destination.Port != 0 {
+				// Just use the first candidate port
+				return destination.Port, nil
```

Comment:
Do you know if the order of these destinations is deterministic?

Reply:
I think if we include `AppGuid` in the `if` check there should only be a single match here.